pkg/network/node/networkpolicy.go

```diff
@@ -789,15 +769,15 @@ func (np *networkPolicyPlugin) parsePeerFlows(npns *npNamespace, npp *npPolicy,
 			if len(peer.NamespaceSelector.MatchLabels) == 0 && len(peer.NamespaceSelector.MatchExpressions) == 0 {
 				// The NamespaceSelector is empty, meaning it selects all namespaces
 				peerFlows = append(peerFlows, "")
-			} else if dir == ingressFlow {
-				// We can implement namespaceSelectors on ingress by just
-				// checking the source VNID...
-				npp.watchesNamespaces = true
-				peerFlows = append(peerFlows, np.selectNamespaces(peer.NamespaceSelector, dir)...)
```

Comment:
Do we not have some way of distinguishing that we are in "migration mode"?

It's _much_ less efficient to use per-IP matches rather than VNID matches. This will break some very large clusters.

Reply:
@danwinship even if we can detect migration and only implement this per IP matches during that time period, do you also foresee issues with large clusters?

Reply:
> Do we not have some way of distinguishing that we are in "migration mode"?

No. We could add an environment variable for that if needed.

But currently per-IP is also used by egress policies, would it get this scale problem?

Reply:
> even if we can detect migration and only implement this per IP matches during that time period, do you also foresee issues with large clusters?

Yes. In theory, some large clusters may have NP-related performance problems during migration that they did not have before migration.

I don't have any good estimates of how large a cluster needs to be to hit this problem though; SDN has always had this particular optimization, so we don't have any evidence of how different customers' clusters would behave without it...

> But currently per-IP is also used by egress policies, would it get this scale problem?

Yes, but egress policies have always worked that way, so people would have already avoided creating egress policies that caused problems in their cluster.

Reply:
I updated the patch. Now, the per-podIP rules will be added after the VNI rules and only when running in migration mode. Also, I take the host-network namespace into account. PTAL